domains/certificates/Query.API/API/Program.cs

```diff
@@ -19,107 +22,126 @@
-builder.Services.AddVersioningToApi();
+        builder.Services.AddActivityLog(options => options.ServiceName = "certificates");
 
-var tokenValidationOptions =
-    builder.Configuration.GetSection(TokenValidationOptions.Prefix).Get<TokenValidationOptions>()!;
-builder.Services.AddOptions<TokenValidationOptions>().BindConfiguration(TokenValidationOptions.Prefix)
-    .ValidateDataAnnotations().ValidateOnStart();
-var b2COptions = builder.Configuration.GetSection(B2COptions.Prefix).Get<B2COptions>()!;
-builder.Services.AddOptions<B2COptions>().BindConfiguration(B2COptions.Prefix).ValidateDataAnnotations()
-    .ValidateOnStart();
-builder.Services.AddB2CAndTokenValidation(b2COptions, tokenValidationOptions);
+        builder.Services.AddQueryApi();
+        builder.Services.AddContractService();
+        builder.Services.AddMeasurementsSyncer();
+        builder.Services.AddIssuingContractCleanup();
+        builder.Services.AddVersioningToApi();
 
+        var tokenValidationOptions =
+            builder.Configuration.GetSection(TokenValidationOptions.Prefix).Get<TokenValidationOptions>()!;
+        builder.Services.AddOptions<TokenValidationOptions>().BindConfiguration(TokenValidationOptions.Prefix)
+            .ValidateDataAnnotations().ValidateOnStart();
+        var b2COptions = builder.Configuration.GetSection(B2COptions.Prefix).Get<B2COptions>()!;
+        builder.Services.AddOptions<B2COptions>().BindConfiguration(B2COptions.Prefix).ValidateDataAnnotations()
+            .ValidateOnStart();
+        builder.Services.AddB2CAndTokenValidation(b2COptions, tokenValidationOptions);
 
-var app = builder.Build();
 
-app.MapHealthChecks("/health");
+        var app = builder.Build();
 
-app.AddSwagger("certificates");
+        app.MapHealthChecks("/health");
 
-app.UseHttpsRedirection();
+        app.AddSwagger("certificates");
 
-app.UseAuthentication();
-app.UseAuthorization();
+        app.UseHttpsRedirection();
 
-app.MapControllers();
+        app.UseAuthentication();
+        app.UseAuthorization();
 
-var activityLogApiVersionSet = app.NewApiVersionSet("activitylog").Build();
-app.UseActivityLog().WithApiVersionSet(activityLogApiVersionSet)
-    .HasApiVersion(ApiVersions.Version20240423AsInt)
-    .HasApiVersion(ApiVersions.Version20230101AsInt);
-app.UseActivityLogWithB2CSupport().WithApiVersionSet(activityLogApiVersionSet).HasApiVersion(ApiVersions.Version20240515AsInt);
+        app.MapControllers();
 
+        var activityLogApiVersionSet = app.NewApiVersionSet("activitylog").Build();
+        app.UseActivityLog().WithApiVersionSet(activityLogApiVersionSet)
+            .HasApiVersion(ApiVersions.Version20240423AsInt)
+            .HasApiVersion(ApiVersions.Version20230101AsInt);
+        app.UseActivityLogWithB2CSupport().WithApiVersionSet(activityLogApiVersionSet)
+            .HasApiVersion(ApiVersions.Version20240515AsInt);
 
-app.Run();
 
-public partial class Program
-{
+        if (args.Contains("--swagger"))
```

Comment:
Is this code needed?

Swashbuckle+dotnet has built-in support for generating swagger spec on build. I think I showed you two links explaining how to do it, by modifying the ".csproj" file.

Reply:
This is an easy way to handle multiple versions of the API